app/models/manageiq/providers/openstack/network_manager/security_group.rb

```diff
@@ -55,6 +83,42 @@ def delete_security_group_queue(userid)
     MiqTask.generic_action_with_callback(task_opts, queue_opts)
   end
 
+  def delete_security_group_rule_queue(userid, key)
+    task_opts = {
+      :action => "delete Security Group rule for user #{userid}",
+      :userid => userid
+    }
+    queue_opts = {
+      :class_name  => self.class.name,
+      :method_name => 'raw_delete_security_group_rule',
+      :instance_id => id,
+      :priority    => MiqQueue::HIGH_PRIORITY,
+      :role        => 'ems_operations',
+      :zone        => ext_management_system.my_zone,
+      :args        => [key]
+    }
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def raw_create_security_group_rule(security_group_id, direction, options)
+    options.delete_if { |_k, v| v.nil? || v.empty? }
```

Comment:
You're right, usually Fog doesn't care about other arguments besides the one it handles.
Meanwhile if a managed argument is empty or nil that can break the API request therefore the cleansing.
That operation cannot be done before reaching the provider because they all have different argument needs.

Also, there are actually some cleaning done in other places such as [1] and [2]

[1] https://github.com/ManageIQ/manageiq-providers-openstack/blob/master/app/models/manageiq/providers/openstack/network_manager/security_group.rb#L21
[2] https://github.com/ManageIQ/manageiq-providers-openstack/blob/master/app/models/manageiq/providers/openstack/network_manager/network_router.rb#L12-L13